apps/bfd-server/bfd-server-war/src/main/webapp/WEB-INF/web.xml

```diff
@@ -6,12 +6,29 @@
 	<filter>
 		<filter-name>RequestResponseLoggingFilter</filter-name>
 		<filter-class>gov.cms.bfd.server.war.RequestResponseLoggingFilter</filter-class>
+		<async-supported>true</async-supported>
```

Comment:
Curious if folks have a greater understanding of how indicating that async is supported will impact the requests as they travel upstream (further from the client). This was only required on the downstream filter here and not on the filter or servlet upstream of the `UserAwareDoSFilter`.

Reply:
From the servlet spec it is permitted for an asynchronous filter/servlet to dispatch to a synchronous filter/servlet but the other way (sync to async) is not permitted. So I think that the upstream filter (the metrics filter) is fine to be left as-is without adding async-supported. Looking at that metrics filter code it looks like it does actually support async but it does not itself make a call to startAsync() so I don't think there is value unless something further along in the chain starts to take advantage of asynchronous processing.

As far as adding this to the logging filter, it looks like it will be necessary if we want to keep the logging filter as the first filter in the chain (and that seems correct). However, I think we may need to write some code in the logging filter to handle the case when the DosFilter delays a request by adding an async listener (similar to what you can find in the metrics abstract filter class). In your testing, did the logging filter perform its response actions correctly on the delayed requests?

Reply:
That's excellent to know—thanks for doing the work of reading the spec. Haven't had the chance yet, but I'm going to look through for my own education.

> I think we may need to write some code in the logging filter to handle the case when the DosFilter delays a request by adding an async listener

So the behavior I saw was that if `RequestResponseLoggingFilter` didn't have the `async-supported` configured, then it stopped logging anything on incoming requests. After changing the config, logging _seemed_ to resume as normal, but I need to go back and double check that all of the fields are added.

Your question is making me worry (and it would make sense) that it's just running `handleRequest` and not `handleResponse`. 

Will get back with a more definitive answer next week, unless you get bored!

Reply:
I played around with it a little bit and it appears that when the DosFilter delays a request we are getting two entries in the access.json -- one that is partially formed and one that is fully formed. I think that the partially formed one is the result of the original thread and needs to be suppressed (or perhaps just cleaned up to make it clear as to what it is) by putting in some listener logic.